pkg/ui/react-app/src/thanos/pages/stores/StorePoolPanel.tsx

```diff
@@ -16,13 +16,25 @@ export const columns = [
   'Endpoint',
   'Status',
   'Announced LabelSets',
-  'Min Time',
-  'Max Time',
+  'Min Time (UTC)',
+  'Max Time (UTC)',
   'Last Successful Health Check',
   'Last Message',
 ];
 
-export const MAX_TIME = 9223372036854775807;
+export const storeTimeRangeMsg = (validMin: boolean, validMax: boolean): string => {
+  if (!validMin && !validMax) {
```

Comment:
I think it is important to distinguish between the cases:

* Time range is `MIN_TIME; MAX_TIME` i.e. when Prometheus still hasn't uploaded any blocks. In this case, Query will always send all queries to that StoreAPI; 
* Time range is `MAX_TIME; MIN_TIME` i.e. when Thanos Store has no blocks. In this case, Query will not send any queries to it.

As is right now, both stores would show up with minuses in the Store page and with the same tooltip. I suggest at the minimum to change this function to distinguish between these two states so that it would be clear what's happening. Perhaps something like:
"This store's time range data is not available but we will send all queries to it"? Or maybe we could make the minuses different colors depending on the state? For example, the minuses could be green if we will still send all queries to it. Or perhaps the node's address could have a different color depending on the state e.g. green = we will send all queries to it, yellow = we will send certain queries to it depending on the labels & min/max time, red = no queries will be sent to it?

WDYT @matej-g ?

Reply:
Hm, I'm wondering whether min / max time ranges and the information whether we are querying the store are two separate things. For the min / max, I would expect simply to have the information stated in the dashboard if it is available - if not, regardless of the reason (i.e. no block uploaded or no block in store), we just signal that this information cannot be showed because it is unknown.

But I think you're onto something with color coding the nodes (perhaps we could even add another column, besides `Status` and add details on querying?), although it feels like a separate feature.

Reply:
ping @GiedriusS, how do you suggest to proceed? Do you agree with the suggestion and if so, should we do this in the current PR?

Reply:
Mhm, I guess you are correct. Whether or not we are sending queries to a node is different information altogether and we could work on this in future pull requests. :+1: